File: projects/10/java/Penguin.java
/* Penguin.java provides Penguin  subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Sinai Park
 * Date: April 24 2020
 ******************************************************/

public class Penguin extends WalkingBird {
    
        
     /* explicit constructor
      * Receive: name, a String 
      * PostCond: myName == name.
      */
    public Penguin(String name)
    {
        super(name);
    }
    
    
     /* A Penguin's Call
      * Return: a Penguin-call ("huh-huh-huuuh!").
      */
    public String call()
    {
        return "Huh-huh-huuuh!";
    }
    
    }
    
    

File: projects/10/ruby/Penguin.rb
# Penguin.rb | Defines the Penguin class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Sinai Park
# Date: APril 24 2020
####################################################

require './WalkingBird.rb'

class Penguin < WalkingBird

######################################################
# call returns the sound that a Penguin type makes       #
# Recieve: NONE                                      #
# Return: Huh-huh-huuuh!, a String                           #
######################################################
def call
	"Huh-huh-huuuh!"
end

end